src/packages/auth/functions.ts

```diff
@@ -178,3 +178,77 @@ export async function session(authorizationHeader: string | undefined): Promise<
     return { message: jwtErrorMessage, success: false };
   }
 }
+
+export function validateGroupRoleMappings() {
+  const { DEFAULT_ROLE, AUTH_GROUP_ROLE_MAPPINGS } = getEnv();
+
+  for (const group in AUTH_GROUP_ROLE_MAPPINGS) {
+    // compute intersection of this mapping's roles and DEFAULT_ROLE list
+    // the mapping is invalid if we don't have any overlap, since we can't compute
+    // a default role for this group -> role mapping
+    const roles = new Set(AUTH_GROUP_ROLE_MAPPINGS[group]);
+    const intersection = DEFAULT_ROLE.filter(e => roles.has(e));
+
+    if (intersection.length == 0) {
+      throw new Error(`
+        No roles within DEFAULT_ROLE list were found in the group to role mapping.
+            DEFAULT_ROLE: ${DEFAULT_ROLE}
+            Group: ${group}
+            Roles: ${[...roles]}
+        Roles must share at least one role with DEFAULT_ROLE, which becomes
+        the default role for any user logging in under this group
+      `);
+    }
+  }
+}
+
+export function getDefaultRoleForAllowedRoles(allowedRoles: string[]): string {
+  const { DEFAULT_ROLE } = getEnv();
+  const roles = new Set(allowedRoles);
+
+  for (const defaultRole of DEFAULT_ROLE) {
+    if (roles.has(defaultRole)) {
+      return defaultRole;
+    }
+  }
+
+  throw new Error(`Fatal error, not able to find a matching default role within the following auth roles: ${allowedRoles}`);
+}
+
+export function mapGroupsToRoles(groupList: string[]): UserRoles {
+  const { DEFAULT_ROLE, ALLOWED_ROLES, AUTH_GROUP_ROLE_MAPPINGS } = getEnv();
+
+  let default_role = DEFAULT_ROLE[0];
+  let allowed_roles  = ALLOWED_ROLES;
+
+  // use auth group -> aerie role mappings if set
+  if (JSON.stringify(AUTH_GROUP_ROLE_MAPPINGS) !== '{}') {
+    const mappedGroupMembership = getGroupsWithMappings(groupList);
+    if (mappedGroupMembership.length > 0) {
+      allowed_roles = getAllAllowedRolesForAuthGroups(mappedGroupMembership);
+      default_role = getDefaultRoleForAllowedRoles(allowed_roles);
+    }
+  }
+
+  return {
+    allowed_roles,
+    default_role
+  };
+}
+
+export function getGroupsWithMappings(authGroups: string[]): string[] {
+  const { AUTH_GROUP_ROLE_MAPPINGS } = getEnv();
+  const authGroupsSet = new Set(authGroups);
+
+  return Object
+    .keys(AUTH_GROUP_ROLE_MAPPINGS)
+    .filter(mappedGroup => authGroupsSet.has(mappedGroup));
+}
```

Comment:
Minor, nonblocking: could you please comment this function? It took a couple readthroughs to realize that this was taking in a set of groups and removing the ones that Aerie doesn't have role mappings for. Either a one-liner at the top or inlines like you did for `getAllAllowedRolesForAuthGroups` would be sufficient.